image/helpers.go

```diff
@@ -209,13 +210,9 @@ func NewToolingStore() (*ToolingStore, error) {
 	return newToolingStore(arch, storeID)
 }
 
-// DownloadOptions carries options for downloading snaps plus assertions.
-type DownloadOptions struct {
+// DownloadSnapOptions carries options for downloading snaps plus assertions.
+type DownloadSnapOptions struct {
```

Comment:
Ouch! Just noticed that you already have `SnapDownloadOptions` below :-) And it looks like you named the struct this way because `DownloadSnapOptions` represents the options for the `DownloadSnap()` function, in the same way as `DownloadManyOptions` carries the options for `DownloadMany()`.

I still find it slightly confusing to have these two structures with such a close name. Maybe this one could be called `SnapDownloadOptionsInternal`, if it's not used outside of this file?

Reply:
Or even `SnapDownloadInfo`?

Anyway, I forgot to say it explicitly: the namings you have for the structures is perfectly consistent, since it follows the function names. It's just that IMHO it's better to describe the structs according to what data they hold, since function names could change in the future (or the same struct could be used in different functions).

Reply:
We use <Func>Options naming quite consistently across many places. wrappers has a lot of examples. These are structs meant to be used with the functions, so one should really start from the function signature. We use them partly because go doesn't have named function arguments.

About trying to find a name unrelated to the functions, I disagree, is worth the effort only once we actually want to reuse the struct across more than one function.

Anyway the naming around DownloadMany is quite close to the one of:

wrappers.EnsureSnapServices(snaps map[*snap.Info]*SnapServiceOptions, opts *EnsureSnapServicesOptions, ...)